pkg/metrics/features/metrics.go

```diff
@@ -256,4 +556,107 @@ func (m Metrics) update(params enabledFeatures, config *option.DaemonConfig) {
 
 	deviceMode := config.DatapathMode
 	m.DPDeviceMode.WithLabelValues(deviceMode).Add(1)
+
+	if config.EnableHostFirewall {
+		m.NPHostFirewallEnabled.Add(1)
+	}
+
+	if config.EnableLocalRedirectPolicy {
+		m.NPLocalRedirectPolicyEnabled.Add(1)
+	}
+
+	if params.IsMutualAuthEnabled() {
+		m.NPMutualAuthEnabled.Add(1)
+	}
+
+	if config.EnableNonDefaultDenyPolicies {
+		m.NPNonDefaultDenyEnabled.Add(1)
+	}
+
+	for _, mode := range config.PolicyCIDRMatchMode {
+		m.NPCIDRPoliciesToNodes.WithLabelValues(mode).Add(1)
+	}
+
+	if config.EnableIPSec {
+		if config.EncryptNode {
+			m.ACLBTransparentEncryption.WithLabelValues(advConnNetEncIPSec, "true").Add(1)
+		} else {
+			m.ACLBTransparentEncryption.WithLabelValues(advConnNetEncIPSec, "false").Add(1)
+		}
+	}
+	if config.EnableWireguard {
+		if config.EncryptNode {
+			m.ACLBTransparentEncryption.WithLabelValues(advConnNetEncWireGuard, "true").Add(1)
+		} else {
+			m.ACLBTransparentEncryption.WithLabelValues(advConnNetEncWireGuard, "false").Add(1)
+		}
+	}
+
+	if config.KubeProxyReplacement == option.KubeProxyReplacementTrue {
+		m.ACLBKubeProxyReplacementEnabled.Add(1)
+	}
+
+	if config.NodePortAcceleration != option.NodePortAccelerationDisabled {
+		m.ACLBStandaloneNSLB.WithLabelValues(config.NodePortMode, config.NodePortAlg, config.NodePortAcceleration).Add(1)
+	}
+
+	if config.BGPAnnouncePodCIDR || config.BGPAnnounceLBIP {
```

Comment:
This is from metallb bgp, we have deprecated this code and I intend to remove this config and metallb code from Cilium soon ( before 1.17 feature freeze ). I'd say we should not add `BGPAnnouncePodCIDR` and `BGPAnnounceLBIP` related metrics.

Reply:
Discussed offline, let's keep those flags and add `EnableBGPControlPlane` flag. I will remove metallb specific config flags when actual metallb code removal PR is created.